### ISSUE TITLE
Bindings Http Client proposal

### ISSUE THREAD

**Issue by CarlosLanderas** [dapr/dotnet-sdk#230]
I can't see the bindings http client in the Dapr.Client. Looking at the source code I see we have available State, Invoke and Publish but not bindings. 

I find it would be very interesting to extend this functionality so we do not have to be working with Http Client Directly

I've seen the go sdk has it:

```go
_, err = client.InvokeBinding(context.Background(), &pb.InvokeBindingEnvelope{
		Name: "storage",
		Data: &any.Any{
			Value: []byte("some data"),
		},
	})
```

I can create a pull request with this functionality if you considering interesting.

I thought about the signature and I suppose it should be something like this:

```csharp
   /// <summary>
    /// A client for interacting with the Dapr bindings endpoints.
    /// </summary>x
    public abstract class BindingsClient
    {
        /// <summary>
        /// Invokes the specified binding on target service, Request data is serialized using the default System.Text.Json Serialization.
        /// </summary>
        /// <param name="bindingName">The name of the binding to be called in the Dapr binding request.</param>
        /// <param name="data">The data to be sent within the body of the Dapr invoke request.</param>
        /// <param name="metadata">Specific metadata to be sent to the binding endpoint.</param>
        /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
        /// <typeparam name="TRequest">The data type of the object that will be serialized.</typeparam>
        /// <returns>A <see cref="Task" />.</returns>
        public abstract Task InvokeBindingAsync<TRequest>(string bindingName, TRequest data, Dictionary<string,string> metadata = null, CancellationToken cancellationToken = default);
    }
```

If you give me the go I start working on it.

Thanks!

**Comment by amanbha** [comment]
@CarlosLanderas  We are int he process of refactoring the client sdk. The sdks will nto have independent clients, instead there will be one client with multiple methods on it. The SDK underneath will use auto-generated gRPC. 
Work item is tracked in following:
 https://github.com/dapr/dotnet-sdk/issues/223
https://github.com/dapr/dotnet-sdk/issues/197

**Comment by CarlosLanderas** [comment]
@amanbha thank you for the information!. I've already created the bindings http client (I need it for a talk this thursday) but what you mention makes sense. I'm glad @rynowak is involved here. He is a super master :)

Thank you!

Closing!

**Comment by CarlosLanderas** [comment]
PD: @amanbha when this unified client will be available? Do you find interesing having this client in the meantime, or shall I close the issue

**Comment by amanbha** [comment]
@CarlosLanderas The new integrated client will be available with our next release 0.5.0 slated for March end. Givne that it will be available in next release, you can close this issues.
